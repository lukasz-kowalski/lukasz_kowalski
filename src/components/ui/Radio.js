import React from 'react';
import withHandlers from './withHandlers';

const Radio = ({ radioData, id, label, isRequired, errorMsg, handleChange, isValid, reference, value }) => (
  <React.Fragment>
    <label htmlFor={id}>
      {label}{isRequired && <span className="blue">*</span>}
    </label>
    <div className="form-check" id={id}>
      {radioData && radioData.map((item, index) =>
        <React.Fragment key={item.id + index}>
          <input 
            className={`form-check-input ${!isValid ? 'is-invalid' : ''}`}
            type="radio"
            name={id} 
            id={item.id} 
            value={item.value}
            ref={index === 0 ? reference : undefined}  // set focus on first radio button
            checked={value === item.value}
            onChange={handleChange}
          />
          <label className="form-check-label radio-label" htmlFor={item.id}>
            {item.label}
          </label>
        </React.Fragment>
      )}
      <div className="invalid-feedback">
        {errorMsg}
      </div>
    </div>
  </React.Fragment>
);

export default withHandlers(Radio);
