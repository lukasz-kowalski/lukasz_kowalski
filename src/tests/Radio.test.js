import React from 'react';
import { shallow } from 'enzyme';
import Radio from '../components/ui/Radio';

let wrapper;

const radioData = [
  {
    id: 'character-male',
    value: 'male',
    label: 'Male'
  },
  {
    id: 'character-female',
    value: 'female',
    label: 'Female'
  },
  {
    id: 'character-na',
    value: 'n/a',
    label: 'n/a'
  }
];

beforeEach(() => {
  wrapper = shallow(
  <Radio
    id="character-gender"
    radioData={radioData}
  />);
})

test('should successful render Radio', () => {
  expect(wrapper).toMatchSnapshot();
})
