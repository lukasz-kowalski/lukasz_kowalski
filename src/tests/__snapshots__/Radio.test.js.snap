// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successful render Radio 1`] = `
<Radio
  handleChange={[Function]}
  id="character-gender"
  isValid={true}
  radioData={
    Array [
      Object {
        "id": "character-male",
        "label": "Male",
        "value": "male",
      },
      Object {
        "id": "character-female",
        "label": "Female",
        "value": "female",
      },
      Object {
        "id": "character-na",
        "label": "n/a",
        "value": "n/a",
      },
    ]
  }
  validate={[Function]}
  value=""
/>
`;
